### DISCUSSION TITLE
Notifications problem - electron.app.superProductivity

### DISCUSSION THREAD

**Discussion by thanksmonster** [super-productivity/super-productivity#3497] [Q&A] [answered]
I want to report a possible bug related to the notifications. Yesterday I accidentally turned off superProductivity notifications: 

![image](https://github.com/user-attachments/assets/08f55a94-5519-43e0-9905-6f00e12320cb)

And with that there is no way (so far) of reactivating it. I've tried:

- Reinstalled the app multiple times (I even restarted it a few times)
- Removed all registry entries (related to superProductivity string)
- Removed the local installation folder
- I also tried to import regedit entries from another computer `Computer\HKEY_CURRENT_USER\SOFTWARE\Microsoft\Windows\CurrentVersion\Notifications\Settings\electron.app.superProductivity`

I was able to reproduce the error in another virtual machine. I'm using Pomodoro timer/notifications for testing.

Some extra info:

OS: Windows 10 Pro 22H2
superProductivity: 10.0.4

is there anything else I can do?

Thanks a lot.

**Comment by johannesjo** [comment, marked as answer]
**!!! Disclaimer: I am not using windows at the moment, so please forgive me, for just posting what chatgpt spat out (not tested!!!):**


If notifications for your app have stopped working on Windows after being disabled, you can try the following steps to re-enable and troubleshoot notifications:

### 1. **Re-enable Notifications in Windows Settings:**
   - **Open Windows Settings**:
     - Press `Windows Key + I` to open the **Settings** app.
   - **Go to Notifications Settings**:
     - Navigate to **System** > **Notifications & actions**.
   - **Enable Notifications for Your App**:
     - Scroll down to the section **Get notifications from these senders**.
     - Find your app in the list, and toggle the switch **On**.
     - If your app is not listed, reinstalling the app or restarting your system may force Windows to recognize it again.

### 2. **Check Focus Assist Settings:**
   - **Open Focus Assist**:
     - Focus Assist is a Windows feature that can block notifications during certain times or activities.
     - In **Settings**, go to **System** > **Focus assist**.
   - **Disable Focus Assist**:
     - Make sure Focus Assist is set to **Off**, or configure it so notifications from your app are allowed when it's on.

### 3. **Check App Notification Settings (if available):**
   - **Open the App's Settings**:
     - Some apps have their own internal notification settings.
     - Check the settings inside the app to ensure notifications are enabled.
   - **Reinstall the App**:
     - If the app has a bug, reinstalling it may reset the notification system.

### 4. **Reset Notifications Using PowerShell (Advanced):**
   You can reset notifications for your app by deleting its notification cache using PowerShell:
   - Press `Windows Key + X`, and select **Windows PowerShell (Admin)**.
   - Run the following command to reset all notifications:
     ```bash
     Get-Process explorer | Stop-Process
     Remove-Item "$env:APPDATA\Microsoft\Windows\Notifications\wpndatabase.db"
     Start-Process explorer.exe
     ```

### 5. **Check Windows Notifications and Maintenance:**
   - **Open Action Center**:
     - Press `Windows Key + A` to open the Action Center.
     - Verify that notifications for the app were not accidentally dismissed or silenced here.

### 6. **Restart the Windows Notification Service:**
   - Open **Services** (type `services.msc` in the Start Menu).
   - Find **Windows Push Notifications User Service**.
   - Right-click it and choose **Restart** to refresh notification services.

### 7. **Update Windows and Your App:**
   - Make sure both Windows and your app are fully updated, as this can resolve issues caused by outdated software.

If none of the above work, there may be an app-specific bug or a deeper system issue requiring further investigation.

**Comment by thanksmonster** [comment]
Dear @johannesjo 

Thanks a lot! Clearing the wpndatabase.db did the trick. In my case i had to stop the Windows push notification service first:

1. Press Windows Key + X, and select Windows PowerShell (Admin).
2. `Get-Service -DisplayName  "Windows Push Notification*" | Stop-Service`
3. `Remove-Item "C:\Users\**USERNAME**\AppData\Local\Microsoft\Windows\Notifications\wpndatabase.db"` (or just rename the file to wpndatabase.db.old )
4. `Get-Service -DisplayName  "Windows Push Notification*" | Start-Service`
5. Restart the computer

Thanks again!
